WcaOnRails/app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -8,6 +8,40 @@ class Api::V0::UserRolesController < Api::V0::ApiController
     end
   end
 
+  private def status_sort_rank(status)
+    case status
+    when 'leader'
+      1
+    when 'senior_member'
+      2
+    when 'member'
+      3
+    else
+      4
+    end
+  end
+
+  # Sorts the list of roles based on the given list of sort keys and directions.
+  private def sorted_roles(roles, sort_param)
+    # The value of sort_param is inspired from https://specs.openstack.org/openstack/api-wg/guidelines/pagination_filter_sort.html.
+    sort_param ||= ''
+    sort_keys_and_directions = sort_param.split(',')
+    roles.sort_by { |role|
+      sort_keys_and_directions.map { |sort_key_and_direction|
```

Comment:
Are you sure that Ruby's `sort_by` is stable? I vaguely recall that it isn't, although I'm not 100% sure if I remember correctly.

Reply:
Turns out, the sorting _is_ unstable even in the latest Ruby releases: https://ruby-doc.org/3.3.0/Array.html#method-i-sort

> When the [comparison callback] returns zero, the order for a and b is indeterminate, and may be unstable: